src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +108,311 @@ const ActorInit En_Pst_InitVars = {
     (ActorFunc)EnPst_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80B2C4B8 = {
-    { COLTYPE_HIT1, AT_NONE, AC_NONE, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK1, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_NONE, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_HIT1,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 28, 72, 0, { 0, 0, 0 } },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit2 D_80B2C4E4 = { 1, 0, 0, 0, MASS_IMMOVABLE };
+static CollisionCheckInfoInit2 sColChkInfoInit = { 1, 0, 0, 0, MASS_IMMOVABLE };
+
+static AnimationInfoS sAnimationInfo[] = { &object_pst_Anim_000018, 1.0f, 0, -1, ANIMMODE_ONCE, 0 };
+
+void EnPst_UpdateCollision(EnPst* this, PlayState* play) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
+
+s32 func_80B2B874(EnPst* this) {
+    switch (this->actor.params) {
+        case 0:
+            return gSaveContext.save.weekEventReg[27] & 0x2;
+        case 1:
+            return gSaveContext.save.weekEventReg[27] & 0x4;
+        case 2:
+            return gSaveContext.save.weekEventReg[27] & 0x8;
+        case 3:
+            return gSaveContext.save.weekEventReg[27] & 0x10;
+        case 4:
+            return gSaveContext.save.weekEventReg[27] & 0x20;
+        default:
+            return false;
+    }
+}
+
+s32 func_80B2B8F4(EnPst* this) {
+    switch (this->actor.params) {
+        case 0:
+            return gSaveContext.save.weekEventReg[27] & 0x40;
+        case 1:
+            return gSaveContext.save.weekEventReg[27] & 0x80;
+        case 2:
+            return gSaveContext.save.weekEventReg[28] & 0x1;
+        case 3:
+            return gSaveContext.save.weekEventReg[28] & 0x2;
+        case 4:
+            return gSaveContext.save.weekEventReg[28] & 0x4;
+        default:
+            return false;
+    }
+}
```

Comment:
Seems like these two are to do with depositing letters and the Postman collecting them.